errors.go

```diff
@@ -70,3 +70,17 @@ func (t TypeError) Error() string {
 func NewTypeError(expected, actual interface{}) error {
 	return TypeError{fmt.Sprintf("Invalid type for arguments. Expected: %T, actual: %T", expected, actual)}
 }
+
+type CanceledError struct {
+	p string
```

Comment:
That means that `Error()` will always do the work of rebuilding the string. Seems unnecessary if we can just do the work once.